jiant/tasks/tasks.py

```diff
@@ -3731,3 +3732,121 @@ def get_metrics(self, reset=False):
         """Get metrics specific to the task"""
         acc = self.scorer1.get_metric(reset)
         return {"accuracy": acc}
+
+
+@register_task("wikipedia_corpus_sop", rel_path="wikipedia_corpus_small")
+class SentenceOrderTask(PairClassificationTask):
+    """ Task class for Sentence Order Prediction (SOP) with wikipedia data.
+        See the ALBERT paper for details on SOP: https://arxiv.org/abs/1909.11942.
+        We are currently using an unpreprocessed version of the Wikipedia corpus
+        that consists of 5% of the data. You can generate the data by following the
+        instructions from jiant/scripts/mlm.
+    """
+
+    def __init__(self, path, max_seq_len, name, **kw):
+        super(SentenceOrderTask, self).__init__(name, n_classes=2, **kw)
+        self.path = path
+        self.max_seq_len = max_seq_len
+
+        self.train_data_text = None
+        self.val_data_text = None
+        self.test_data_text = None
+        self.files_by_split = {
+            "train": os.path.join(path, "train.txt"),
+            "val": os.path.join(path, "valid.txt"),
+            "test": os.path.join(path, "test.txt"),
+        }
+
+    def get_data_iter(self, path):
+        """Loading data file and tokenizing the text
+        Args:
+            path: (str) data file path
+        """
+        count = 0
+        with open(path, encoding="utf-8") as txt_fh:
+
+            for row in txt_fh:
+                count += 1
+                if count == 10:
+                    break
+                toks = row.strip()
+                sentences = row.split(".")
+                if len(sentences) <= 1:
+                    continue
+                else:
+                    for i in range(len(sentences) - 1):
+                        if random.uniform(0, 1) > 0.5:
+                            is_right_order = 1
+                            sent_a = sentences[i]
+                            sent_b = sentences[i + 1]
+                        else:
+                            is_right_order = 0
+                            sent_a = sentences[i + 1]
+                            sent_b = sentences[i]
+                        sent_a_processed = tokenize_and_truncate(
+                            self._tokenizer_name, sent_a, self.max_seq_len // 2
+                        )
+                        sent_b_processed = tokenize_and_truncate(
+                            self._tokenizer_name, sent_b, self.max_seq_len // 2
+                        )
+                        yield (sent_a_processed, sent_b_processed, is_right_order)
+
+    def load_data(self):
+        pass
+
+    def process_split(
+        self, split, indexers, model_preprocessing_interface
+    ) -> Iterable[Type[Instance]]:
+        """Process a sentence order prediction split by indexing and creating fields.
```

Comment:
Clarify why this needs to override `process_split`, when most pair tasks don't. I assume it has something to do with making sure the label names match what's loaded from pretrained ALBERT? (This kind of information should be helpful for future cleanup of the structure of tasks...)